src/components/ModalFlows/StakeScreens/StakeScreens.js

```diff
@@ -10,9 +12,26 @@ const ZERO_BN = new BigNumber(toDecimals('0', 18))
 function StakeScreens({ mode, stakeManagement, stakeActions }) {
   const [transactions, setTransactions] = useState([])
   const { accountBalance } = useAppState()
+  const history = useHistory()
+  const { allowance: stakingAllowance } = stakeManagement?.staking || {}
 
   const temporatyTrx = useRef([])
 
+  const handleCreateProposal = useCallback(() => {
+    history.push('/home/create')
+  }, [history])
+
+  const renderOnCompleteActions = useCallback(() => {
```

Comment:
wondering if we should have the button on the modal instead of creating the component here and passsing down to it, thinking about passing only the button props. Or maybe you think on this for the possibility for the modal to show something else than buttons?

Reply:
Yeah i think this approach is a general one where tyou can render anything you'd like.

Also the problem with rendering on the modal is that the modal doesn't control the transactions statuses so it doesn't know when it should render, unless we lift the transaction steps status logic to the modal which don't think makes sense really.